### ISSUE TITLE
Feature: integrate instance agnostic link sharing service

### ISSUE THREAD

**Issue by graphit0** [tgxn/lemmy-explorer#172]
**What do you want to see?**
Integrate https://github.com/RikudouSage/lemmyverse.link / https://lemmyverse.link/ i.e. instance agnostic link sharing service 

**Use Case**
Instance agnostic links simplify sharing content outside of fediverse
Suppose you come across a link to lemmy post on non-native instance. 
Like that https://lemmy.world/post/7284890

How do you open it?
Let's face it, searching in your home instance with post name is very inconvenient
Instance agnostic links address this problem
example of instance agnostic link https://lemmyverse.link/lemmy.world/post/7284890

After clicking this link you can set your home instance and the post will be opened there

**Why**
- improve discoverability of Lemmy Explorer since a lot of links going to be pointing to it
- allow users to rely on home instance they set on Lemmy Explorer while redirected by instance agnostic link

**Comment by tgxn** [comment]
oh yeah i get you, basically a redirection service

like:
- `https://lemmyverse.net/link/lemmy.world/c/world` -> `https://your.home/c/world@lemmy.world
easy, data is in URL

`https://lemmyverse.net/link/lemmy.world/post/7284890` 
don't we need to know the post ID on your local instance for this to work?

**Comment by graphit0** [comment]
> don't we need to know the post ID on your local instance for this to work?

Yes indeed, we need to API query the target/home instance to get post ID. You can see how author does it in linked github: file [LemmyObjectResolver](https://github.com/RikudouSage/lemmyverse.link/blob/master/src/Service/LemmyObjectResolver.php)

I realize that it's much more difficult than simple link reformatting, that's why I proposed integrate his solution if it's possible.

**Comment by tgxn** [comment]
I think community redirection would be doable easily, but maybe the post redirection is a bit out-of-scope, since it would involve realtime scraping of the source and target instance.

**Comment by tgxn** [comment]
https://github.com/RikudouSage/lemmyverse.link
